drift-transport-netty/src/main/java/com/facebook/drift/transport/netty/client/ConnectionPool.java

```diff
@@ -146,15 +155,26 @@ private static void closeConnection(Future<Channel> future)
         });
     }
 
+    private int getNextConnectionId()
+    {
+        nextConnectionId = (nextConnectionId + 1) % connectionConcurrency;
```

Comment:
This can be accessed from every thread fetching a value from the connection pool. It's also going to be shared by threads fetching different connections from the pool so in some scenarios it's not round-robin across connections in the pool it's just random.

So maybe don't use a shared value just use ThreadLocalRandom?

Reply:
+1

Reply:
Right, practically it will be random. I used round-robin to 1)  make unit test deterministic and 2) light weight calculation. I changed to `ThreadLocalRandom` to avoid confusion.